lib/rouge/lexers/sas.rb

```diff
@@ -0,0 +1,568 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class SAS < RegexLexer
+      title "SAS"
+      desc "SAS (Statistical Analysis Software)"
+      tag 'sas'
+      filenames '*.sas'
+      mimetypes *%w(
+        application/x-sas
+        application/x-stat-sas
+        application/x-sas-syntax
+      )
+
+      def self.data_step_statements
+        # from Data step statements - SAS 9.4 Statements reference
+        # http://support.sas.com/documentation/cdl/en/lestmtsref/68024/PDF/default/lestmtsref.pdf
+        @keywords ||= %w(
```

Comment:
We generally use a `Set` for these things for faster lookup.

``` ruby
@data_step_statements ||= Set.new %w(
...
```

should work fine.

Reply:
Thanks, have done as part of https://github.com/rouge-ruby/rouge/pull/1107/commits/01c2cc4ec3f09e7ffb2f2ce7571f6324032d052f